File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.ArrayList;

public class Main {

    public static void push(ArrayList<Integer> stack, int num) {
        stack.add(num);
    }

    public static void top(ArrayList<Integer> stack) {
        int size = size(stack);
        if (size == 0) {
            System.out.println("-1");
        } else {
            size -= 1;
            System.out.println(stack.get(size));
        }
    }

    public static int size(ArrayList<Integer> stack) {
        return stack.size();
    }

    public static void empty(ArrayList<Integer> stack) {
        int size = size(stack);
        if (size == 0) {
            System.out.println("1");
        } else {
            System.out.println("0");
        }
    }

    public static void pop(ArrayList<Integer> stack) {
        int size = size(stack);
        if (size == 0) {
            System.out.println("-1");
        } else {
            size -= 1;
            int popNum = stack.remove(size);
            System.out.println(popNum);
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        ArrayList<Integer> stack = new ArrayList<>();

        int N = Integer.parseInt(br.readLine());
        int count = 0;

        while (count < N) {
            count += 1;
            String input = br.readLine();
            if (input.contains("push")) {
                int num = Integer.parseInt(String.valueOf(input.substring(5)));
                push(stack, num);
            } else if (input.contains("top")) {
                top(stack);
            } else if (input.contains("size")) {
                System.out.println(size(stack));
            } else if (input.contains("empty")) {
                empty(stack);
            } else if (input.contains("pop")) {
                pop(stack);
            }
        }
    }
}

File: 백준/Silver/10828. 스택/스택.py
from sys import stdin as s

#input.txt file
#s = open('input.txt')

n = int(s.readline().rstrip())

stack = []
for i in range(0, n):
    command = list(map(str, s.readline().rstrip().split(" ")))
    if(command[0] == "push"):
        stack.append(command[1])
    elif(command[0] == "pop"):
        if(not len(stack)==0):
            print(stack.pop(-1))
        else:
            print(-1)
    elif(command[0] == "size"):
        print(len(stack))
    elif(command[0] == "top"):
        if(not len(stack)==0):
            print(stack[-1])
        else:
            print(-1)
    elif(command[0] == "empty"):
        if(len(stack) == 0):
            print(1)
        else:
            print(0)
